src/clj/puppetlabs/services/jruby_pool_manager/jruby_core.clj

```diff
@@ -222,18 +222,16 @@
   flush-pool!
   "Flush all the current JRubyInstances and repopulate the pool."
   [pool-context]
-  (jruby-agents/send-flush-and-repopulate-pool! pool-context))
+  (jruby-agents/flush-and-repopulate-pool! pool-context))
 
 (schema/defn ^:always-validate
   flush-pool-for-shutdown!
   "Flush all the current JRubyInstances so that the pool can be shutdown
   without any instances being active."
   [pool-context]
   (let [on-complete (promise)]
-    (log/debug "Beginning flush of JRuby pools for shutdown")
-    (jruby-agents/send-flush-pool-for-shutdown! pool-context on-complete)
-    @on-complete
-    (log/debug "Finished flush of JRuby pools for shutdown")))
+    (jruby-agents/flush-pool-for-shutdown! pool-context on-complete)
```

Comment:
Do you think we still need an `on-complete` promise at this level?  It seems like we could just create/use the promise behind the `jruby-agents/flush-pool-for-shutdown!` function to hold up the call until the flush has been completed and not need to let that percolate up here.